src/com/xilinx/rapidwright/design/AttrSite.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2023, Advanced Micro Devices, Inc.
+ * All rights reserved.
+ *
+ * Author: Chris Lavin, AMD AECG Research Labs.
+ *
+ * This file is part of RapidWright.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.xilinx.rapidwright.design;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import com.xilinx.rapidwright.device.BEL;
+import com.xilinx.rapidwright.device.Site;
+import com.xilinx.rapidwright.device.SiteTypeEnum;
+
+/**
+ * Represents a site and its configurable attributes.
+ * 
+ */
+public class AttrSite {
```

Comment:
What's the reason for this over `SiteAttr` which seems to be more consistent with other classes?

Reply:
Its not a great name, but I think `SiteAttr` is misleading as there are not real attributes being applied to the site, it is just collecting configurations that belong to it. What do you think of `ConfiguredSite`?

Reply:
I like `ConfiguredSite` more than `AttrSite`. My suggestion would be `SiteConfig`, but if you feel more strongly about `ConfiguredSite` then let's go with that.